README.md

```diff
@@ -1,2 +1,233 @@
+      type: joint_state_broadcaster/JointStateBroadcaster
+```
+This controller is found in the `joint_state_broadcaster` package and publishes the state of the robot's joints into ROS directly from Gazebo.
+
+In [09-joints.launch.py](launch/09-joints.launch.py) we also add a ros2_control command via `ExecuteProcess` to start this specific controller.
+
+You can launch this, but its still not quite there.
+
+    ros2 launch urdf_sim_tutorial 09-joints.launch.py
+
+This will run the controller and in fact publish on the `/joint_states` topic....but with nothing in them.
+
+```yaml
+header:
+  stamp:
+    sec: 13
+    nanosec: 331000000
+  frame_id: ''
+name: []
+position: []
+velocity: []
+effort: []
+```
+
+What else do you want Gazebo!? Well, it wants to know more information about the joints.
+
+## ROS 2 Control Joint Definitions
+For every non-fixed joint, we need to add information about the joint in the `ros2_control` tag which tells it what interfaces are supported. Let's start with the head joint. Modify the joint tag in your [URDF](urdf/10-firsttransmission.urdf.xacro#L241) to be the following:
+
+```xml
+    <joint name="head_swivel">
+      <command_interface name="position" />
+      <command_interface name="velocity" />
+      <state_interface name="position"/>
+      <state_interface name="velocity"/>
+    </joint>
+```
+ * Note that the joint name here matches the joint name from the standard URDF `<joint>` tag.
+ * For the moment, let us focus on the `state_interface`s, in which we specify that we want to publish both position and velocity of this joint.
+
+You can run this URDF with our previous launch configuration.
+
+    ros2 launch urdf_sim_tutorial 09-joints.launch.py urdf_package_path:=urdf/10-firsttransmission.urdf.xacro
+
+Now, the head is displayed properly in RViz because the head joint is listed in the joint_states messages.
+
+```yaml
+header:
+  stamp:
+    sec: 4
+    nanosec: 707000000
+  frame_id: ''
+name:
+- head_swivel
+position:
+- -2.9051283156888985e-08
+velocity:
+- 7.575990694887896e-06
+effort:
+- .nan
+```
+
+We could continue adding transmissions for all the non-fixed joints (and we will) so that all the joints are properly published. But, there's more to life than just looking at robots. We want to control them. So, let's get another controller in here.
+
+## Joint Control
+[Here's](config/head.yaml) the next controller config we're adding.
+
+```yaml
+controller_manager:
+  ros__parameters:
+    # ... snip ...
+
+    head_controller:
+      type: position_controllers/JointGroupPositionController
+
+head_controller:
+  ros__parameters:
+    joints:
+      - head_swivel
+    interface_name: position
+```
+
+In English, this is saying to add a new JointGroupPositionController called head_controller, and then, in a new parameter namespace, specify which joints are included and that we are publishing positions. We can do this because we specified `<command_interface name="position" />` in the joint tag.
```

Comment:
```suggestion
In English, this is saying to add a new `JointGroupPositionController`called `head_controller`, and then, in a new parameter namespace, specify which joints are included and that we are publishing positions. We can do this because we specified `<command_interface name="position" />` in the joint tag.
```